uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/WorkPattern.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+use CRM_HRLeaveAndAbsences_BAO_WorkPattern as WorkPattern;
+use CRM_HRLeaveAndAbsences_BAO_ContactWorkPattern as ContactWorkPattern;
+
+/**
+ * Class CRM_HRLeaveAndAbsences_Service_WorkPattern
+ */
+class CRM_HRLeaveAndAbsences_Service_WorkPattern {
+
+  /**
+   * Checks If a work pattern has ever been used by a contact
+   *
+   * @param int $workPatternID
+   *
+   * @return boolean
+   */
+  public function workPatternHasEverBeenUsed($workPatternID) {
+    if ($this->isDefaultWorkPattern($workPatternID) ||
+      $this->workPatternIsLinkedToAContactWorkPattern($workPatternID))
+    {
+      return true;
+    }
+
+    return false;
+  }
+
+  /**
+   * Deletes the WorkPattern with the given id.
+   * Checks first to see if the work pattern can be deleted or not.
+   *
+   * @param int $workPatternID
+   */
+  public function delete($workPatternID) {
+    if ($this->workPatternHasEverBeenUsed($workPatternID)) {
+      throw new UnexpectedValueException('Work pattern cannot be deleted because it is used by one or more contacts');
+    }
+
+    WorkPattern::del($workPatternID);
+  }
+
+  /**
+   * Checks if the work pattern is the default work pattern
+   *
+   * @param int $workPatternID
+   *
+   * @return boolean
+   */
+  private function isDefaultWorkPattern($workPatternID) {
+    $workPattern = WorkPattern::findById($workPatternID);
+
+    if ($workPattern->is_default) {
+      return true;
+    }
+
+    return false;
+  }
+
+  /**
+   * Checks if the work pattern is linked to at least one contact work pattern
+   *
+   * @param int $workPatternID
+   *
+   * @return boolean
+   */
+  private function workPatternIsLinkedToAContactWorkPattern($workPatternID) {
```

Comment:
Let's make this method name a bit shorter. The important thing here is to know if a work pattern is linked to a contact. How it's linked (using a ContactWorkPattern) is not really important. With that said, I believe a better name would be `workPatternIsLinkedToAContact`